language/move-lang/src/hlir/translate.rs

```diff
@@ -521,6 +528,17 @@ fn statement(context: &mut Context, result: &mut Block, e: T::Exp) {
     result.push_back(sp(eloc, stmt_))
 }
 
+fn statement_loop_body(context: &mut Context, body: T::Exp) -> (Block, bool) {
+    let old_has_return_abort = context.has_return_abort;
+    context.has_return_abort = false;
+    let mut loop_block = Block::new();
+    let el = maybe_exp(context, &mut loop_block, None, body);
+    ignore_and_pop(context, &mut loop_block, true, el);
+    let has_return_abort = context.has_return_abort;
+    context.has_return_abort = context.has_return_abort || old_has_return_abort;
+    (loop_block, has_return_abort)
```

Comment:
Can the caller compute the value of `has_return_abort` instead? The dual purpose of creating block and computing this bool makes this function a bit hard to follow.

Reply:
> Can the caller compute the value of has_return_abort instead?

No, I need to update the state correctly, and return the proper value. I'd rather not make the caller also update the context.